### PR TITLE
Create Import status endpoint and dummy ServiceContract

### PR DESCRIPTION
**Pull request by hitarthpattani**
### Description (*)

Create import status endpoint and dummy service contract to provide response details and items response in details.

### Fixed Issues (if relevant)

1. magento/async-import#46: Create Import status endpoint and dummy ServiceContract

### Manual testing scenarios (*)

1. In postman, use [DOMAIN]/rest/V1/import/:uuid URL with GET request. For example http://127.0.0.1/async-import/rest/V1/import/1914670575
2. Define headers to access the API

### Desired Output (*)
{
    "status": "completed",
    "uuid": 1914670575,
    "entity_type": "catalog_product",
    "user_id": 1,
    "user_type": 2,
    "items": [
        {
            "uuid": 1914670575,
            "status": "",
            "serialized_data": "",
            "result_serialized_data": "",
            "error_code": "",
            "result_message": ""
        }
    ]
}

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    const UUID = 'uuid';
+    const ENTITY_TYPE = 'entity_type';
+    const USER_ID = 'user_id';
+    const USER_TYPE = 'user_type';
+    const ITEMS = 'items';
+
+    /**
+     * Get status
+     *
+     * @return string|null
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: We cannot have null in this case, if object exists it always have to have status

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+     * Get error
+     *
+     * @return string|null
+     */
+    public function getError();
+
+    /**
+     * Get uuid
+     *
+     * @return int|null
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: We cannot have null in this case, if object exists it always have to have UUID

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    public function getUuid();
+
+    /**
+     * Get entity type
+     *
+     * @return string|null
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Can not be null

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    public function getEntityType();
+
+    /**
+     * Retrieve current user ID
+     *
+     * @return int|null
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Can not be null

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+     * Retrieve current user type
+     *
+     * @return int|null
+     */
+    public function getUserType();
+
+    /**
+     * Get import items status
+     *
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseItemInterface[]|null
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Can not be null

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    public function getItems();
+
+    /**
+     * @param $status
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setStatus($status);
+
+    /**
+     * @param $error
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setError($error);
+
+    /**
+     * @param $uuid
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setUuid($uuid);
+
+    /**
+     * @param $entityType
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setEntityType($entityType);
+
+    /**
+     * @param $userId
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setUserId($userId);
+
+    /**
+     * @param $userType
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setUserType($userType);
+
+    /**
+     * @param $items
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function setItems($items);
+
+    /**
+     * @param $item
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Please define variable type

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+     * @return \Magento\ImportService\Api\Data\SourceStatusResponseInterface
+     */
+    public function addItem($item);
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: I would avoid of using such method in interface. we can always use setItems.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseItemInterface.php [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ImportService\Api\Data;
+
+/**
+ * Class SourceStatusResponseItem
+ */
+interface SourceStatusResponseItemInterface
```
[the remark this answers is not in the document]
Comment (nuzil) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseItemInterface.php]: Can you please also apply here fixes from 
app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ImportService\Model;
+
+use Magento\ImportService\Api\SourceStatusInterface;
+use Magento\ImportService\Api\Data\SourceStatusResponseInterface;
+use Magento\ImportService\Model\SourceStatusResponseItemFactory;
+use Magento\Authorization\Model\UserContextInterface;
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceStatus.php]: Can you please for now also remove dependency on Authorization module? As we are using microservice approach and should not be dependent on other Magento modules
Its may happen we will need it still but for now I want to have it clean

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+        // Create new response object
+        $response = $this->responseFactory->create();
+
+        try
+        {
+            // Set the required response parameters with appropriate details
+            $response->setStatus(SourceStatusResponseInterface::STATUS_COMPLETED)
+                ->setUuid($uuid)
+                ->setEntityType(Product::ENTITY)
+                ->setUserId($this->userContext->getUserId())
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceStatus.php]: As we removed dependency, please set here for now null value

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+                ->setUserType($this->userContext->getUserType());
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceStatus.php]: As we removed dependency, please set here for now null value

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+            $item = $this->responseItemFactory->create();
+            $item->setUuid($uuid)
+                ->setStatus("")
+                ->setSerializedData("")
+                ->setResultSerializedData("")
+                ->setErrorCode("")
+                ->setResultMessage("");
+
+            // Add sample response item object to response
+            $response->addItem($item);
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceStatus.php]: Please use setItems for now

File: app/code/Magento/ImportService/Model/SourceStatusResponse.php [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+    public function setItems($items)
+    {
+        return $this->setData(self::ITEMS, $items);
+    }
+
+    /**
+     * @param $item
+     * @return $this
+     */
+    public function addItem($item)
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceStatusResponse.php]: Please remove this method

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+ */
+declare(strict_types=1);
+
+namespace Magento\ImportService\Model;
+
+use Magento\ImportService\Api\SourceStatusInterface;
+use Magento\ImportService\Api\Data\SourceStatusResponseInterface;
+use Magento\ImportService\Model\SourceStatusResponseItemFactory;
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Catalog\Model\Product;
```
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed dependency of Product class of Magento_Catalog module from our module.

File: app/code/Magento/ImportService/Model/SourceStatus.php [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        // Create new response object
+        $response = $this->responseFactory->create();
+
+        try
+        {
+            // Set the required response parameters with appropriate details
+            $response->setStatus(SourceStatusResponseInterface::STATUS_COMPLETED)
+                ->setUuid($uuid)
+                ->setEntityType(Product::ENTITY)
```
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed dependency of Product class of Magento_Catalog module from our module.

File: app/code/Magento/ImportService/composer.json [hunk 23]
```diff
@@ -6,6 +6,8 @@
     },
     "require": {
         "php": "~7.1.3||~7.2.0",
+        "magento/module-authorization": "*",
+        "magento/module-catalog": "*",
```
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/composer.json]: Removed dependency of magento/module-catalog module from our module.

File: app/code/Magento/ImportService/etc/module.xml [hunk 24]
```diff
@@ -6,6 +6,10 @@
  */
 -->
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
-    <module name="Magento_ImportService" >
+    <module name="Magento_ImportService">
+    	<sequence>
+            <module name="Magento_Authorization"/>
+            <module name="Magento_Catalog"/>
```
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/etc/module.xml]: Removed dependency of Magento_Catalog module from our module

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php [hunk 25]
```diff
@@ -0,0 +1,123 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ImportService\Api\Data;
+
+/**
+ * Class SourceStatusResponse
+ */
+interface SourceStatusResponseInterface
```
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Changed file name prefix from Source to Import for all relevant files of current module commit.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseItemInterface.php (hunk 15 above)
[the remark this answers is not in the document]
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseItemInterface.php]: Changed file name prefix from `Source `to `Import `for all relevant files of current module commit.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 1 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: remove return type `null `from interface.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 2 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: remove return type `null `from interface.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 3 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: remove return type `null `from interface.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 4 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: remove return type `null `from interface.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 5 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: remove return type `null `from interface.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 6 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 7 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 8 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 10 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 11 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 12 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 13 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Model/SourceStatus.php (hunk 19 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed `AddItem `method from respective class and interface, and replaced `setItems `code.

File: app/code/Magento/ImportService/Model/SourceStatusResponse.php (hunk 20 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatusResponse.php]: Removed `AddItem `method from respective class and interface, and replaced `setItems `code.

File: app/code/Magento/ImportService/Model/SourceStatus.php (hunk 18 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed the dependency of Magento_Authorization module.

File: app/code/Magento/ImportService/Model/SourceStatus.php (hunk 17 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed the dependency of Magento_Authorization module.

File: app/code/Magento/ImportService/Model/SourceStatus.php (hunk 16 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceStatus.php]: Removed the dependency of Magento_Authorization module.

File: app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php (hunk 9 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/SourceStatusResponseInterface.php]: Define variable type for parameters for the respective method.

File: app/code/Magento/ImportService/Api/Data/ImportStatusResponseInterface.php [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Get import items status
+     *
+     * @return \Magento\ImportService\Api\Data\ImportStatusResponseItemInterface[]
+     */
+    public function getItems();
+
+    /**
+     * @param string $status
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/ImportStatusResponseInterface.php]: I know that currently Travis didnt highlight this as an error, but I know if will when we will try to merge.
Can you please add description to all Methods
as you made before
Example:
/**
     * Retrieve current user type
     *
     * @return int
     */

File: app/code/Magento/ImportService/Api/Data/ImportStatusResponseItemInterface.php [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+
+    /**
+     * Get result message
+     *
+     * @return string|null
+     */
+    public function getResultMessage();
+
+    /**
+     * @param $uuid
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/ImportStatusResponseItemInterface.php]: Please define variable types in all annotations in this class

File: app/code/Magento/ImportService/Model/ImportStatus.php [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    private $responseFactory;
+
+    /**
+     * @var ImportStatusResponseItem
+     */
+    private $responseItemFactory;
+
+    /**
+     * Status constructor.
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/ImportStatus.php]: Please add empty line after description

File: app/code/Magento/ImportService/Api/Data/ImportStatusResponseInterface.php (hunk 26 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/ImportStatusResponseInterface.php]: I have added comments to each method that has been created for this comment within interfaces and classes.

File: app/code/Magento/ImportService/Api/Data/ImportStatusResponseItemInterface.php (hunk 27 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Api/Data/ImportStatusResponseItemInterface.php]: I have defined a variable type to all interface annotations and classes created for this commit.

File: app/code/Magento/ImportService/Model/ImportStatus.php (hunk 28 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/ImportStatus.php]: Added new empty line after description

**Pull request merged**